packages/maskbook/src/plugins/Trader/settings.ts

```diff
@@ -82,8 +83,10 @@ export function getCurrentDataProviderGeneralSettings(dataProvider: DataProvider
             return coinGeckoSettings
         case DataProvider.COIN_MARKET_CAP:
             return coinMarketCapSettings
+        case DataProvider.UNISWAP:
+            return coinUniswapSettings
         default:
-            unreachable(dataProvider)
+            unreachable(dataProvider as never)
```

Comment:
Should not use `as never`.

Reply:
vscode complains, does this affect error print?

```
export function unreachable(val: never): never {
    console.error('Unhandled value: ', val)
    throw new Error('Unreachable case:' + val)
}
```

Reply:
This helps throw an error for future data providers was added. For every new provider should impls those functions.

Reply:
got it